Jetchat/app/src/main/java/com/example/compose/jetchat/components/JetchatDrawer.kt

```diff
@@ -210,3 +225,41 @@ fun DrawerPreviewDark() {
         }
     }
 }
+
+@Composable
+private fun WidgetDiscoverability() {
+    val context = LocalContext.current
+    Row(
+        modifier = Modifier
+            .height(56.dp)
+            .fillMaxWidth()
+            .padding(horizontal = 12.dp)
+            .clip(CircleShape)
+            .clickable(onClick = {
+                addWidgetToHomeScreen(context)
+            }),
+        verticalAlignment = CenterVertically
+    ) {
+        Text(
+            stringResource(id = R.string.add_widget_to_home_page),
+            style = MaterialTheme.typography.bodyMedium,
+            color = MaterialTheme.colorScheme.onSurface,
+            modifier = Modifier.padding(start = 12.dp)
+        )
+    }
+}
+
+private fun addWidgetToHomeScreen(context: Context) {
+    val appWidgetManager = AppWidgetManager.getInstance(context)
+    val myProvider = ComponentName(context, WidgetReceiver::class.java)
+    if (widgetAddingIsSupported(context)) {
+        appWidgetManager.requestPinAppWidget(myProvider, null, null)
+    }
+}
+
+@ChecksSdkIntAtLeast(api = Build.VERSION_CODES.O)
+private fun widgetAddingIsSupported(context: Context): Boolean {
+    val appWidgetManager = AppWidgetManager.getInstance(context)
+    return Build.VERSION.SDK_INT >=
+        Build.VERSION_CODES.O && appWidgetManager.isRequestPinAppWidgetSupported
```

Comment:
This way you prevent retrieving the widget manager on SDK < O
```suggestion
    return Build.VERSION.SDK_INT >=
        Build.VERSION_CODES.O && AppWidgetManager.getInstance(context).isRequestPinAppWidgetSupported
```